src/runtime_config.rs

```diff
@@ -0,0 +1,79 @@
+use serde::Deserialize;
+use tokio::sync::RwLock;
+use tracing::info;
+
+#[derive(Debug, Deserialize, Clone)]
+pub struct RuntimeConfig {
+    /// A list of tasks to drop before inserting into sqlite.
+    pub drop_task_killswitch: Vec<String>,
+}
+
+pub struct RuntimeConfigManager {
+    pub config: RwLock<RuntimeConfig>,
+    pub path: String,
+}
+
+impl RuntimeConfigManager {
+    pub fn new(path: String) -> Self {
+        let runtime_config = Self::read_yaml_file(&path);
+        Self {
+            config: RwLock::new(runtime_config),
+            path,
+        }
+    }
+
+    fn read_yaml_file(path: &str) -> RuntimeConfig {
+        let contents = std::fs::read_to_string(path)
+            .unwrap_or_else(|_| panic!("Failed to read config file from {}", path));
+        serde_yaml::from_str(&contents)
+            .unwrap_or_else(|_| panic!("Failed to parse YAML from {}", path))
```

Comment:
We shouldn't panic here. If we push a bad configuration, I would much prefer we emit an error to sentry or logging and ignore the invalid configuration. If we panic, a bad config change will terminate our brokers.

Reply:
Yeah, that'd be quite catastrophic. Updated to log an error instead!